core/vm/logger.go

```diff
@@ -18,12 +18,157 @@ package vm
 
 import (
 	"fmt"
+	"math/big"
 	"os"
 	"unicode"
 
 	"github.com/ethereum/go-ethereum/common"
 )
 
+type Storage map[common.Hash]common.Hash
+
+func (self Storage) Copy() Storage {
+	cpy := make(Storage)
+	for key, value := range self {
+		cpy[key] = value
+	}
+
+	return cpy
+}
+
+// StructuredLogCollector is the basic interface to capture emited logs by the EVM logger.
+type StructuredLogCollector interface {
+	// Adds the structured log to the collector.
+	AddStructuredLog(StructLog)
+}
```

Comment:
I do like the fact that log collection is an interface.

The naming is weird though. I think you should go for either `StructuredLog` or `StructLog`
and be consistent about it. I'd personally call the method `StructLog` and the type `StructLogger` because
they are shorter. Note that no interface ambiguity can arise because the parameter is `vm.StructLog`.